### PR TITLE
DCS-549 Restricting to specific case sampling role

### PR DESCRIPTION
**Pull request by andrewrlee**
also allow passing through buffer percentage

### CODE REVIEW

File: src/test/java/uk/gov/justice/hmiprobation/casesampler/controllers/SampleControllerIntegrationTest.kt [hunk 1]
```diff
@@ -53,11 +54,32 @@ class SampleControllerIntegrationTest(
...
+                          "officer" : "Ned"
+                        }]
+                    """".trimIndent()
+
     @BeforeEach
     fun setUp() {
-        Mockito.reset(caseListService)
+        Mockito.reset(caseSamplingService)
 
-        `when`(caseListService.process(any(), any(), any())).thenReturn(PrimaryCaseSampleProvisional(
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmiprobation/casesampler/controllers/SampleControllerIntegrationTest.kt:60]: if using mockitokotlin2 then better to do a `whenever`

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/test/java/uk/gov/justice/hmiprobation/casesampler/controllers/SampleControllerIntegrationTest.kt (hunk 1 above)
Comment (andrewrlee) [reply src/test/java/uk/gov/justice/hmiprobation/casesampler/controllers/SampleControllerIntegrationTest.kt:60]: not currently using that, I was using "com.ninja-squad:springmockk:2.0.1" but couldn't get it working with spring tests. It used to just ignore setting up stubbing on `mockBean`s

**Pull request merged**